src/SimpleNotePad.java

```diff
@@ -1,120 +1,43 @@
 import javax.swing.JFrame;
 import javax.swing.JMenu;
 import javax.swing.JMenuBar;
-import javax.swing.JMenuItem;
-import javax.swing.JOptionPane;
 import javax.swing.JScrollPane;
 import javax.swing.JTextPane;
-import javax.swing.text.Position;
-import javax.swing.text.StyledDocument;
-public class SimpleNotePad extends JFrame implements ActionListener{
+
+public class SimpleNotePad extends JFrame {
     JMenuBar mb = new JMenuBar();
-    JMenu fm = new JMenu("File");
-    JMenu em = new JMenu("Edit");
-    JTextPane d = new JTextPane();
-    JMenuItem nf = new JMenuItem("New File");
-    JMenuItem sf = new JMenuItem("Save File");
-    JMenuItem pf = new JMenuItem("Print File");
-    JMenuItem u = new JMenuItem("Undo");
-    JMenuItem c = new JMenuItem("Copy");
-    JMenuItem p = new JMenuItem("Paste");
+    JMenu filesMenu = new JMenu("File");
+    JMenu recentFilesMenu = new JMenu("Recent");
+    JMenu editMenu = new JMenu("Edit");
+    JTextPane textPane = new JTextPane();
+
+    RecentFilesManager recentFilesManager = new RecentFilesManager(recentFilesMenu);
+
     public SimpleNotePad() {
         setTitle("A Simple Notepad Tool");
-        fm.add(nf);
-        fm.addSeparator();
-        fm.add(sf);
-        fm.addSeparator();
-        fm.add(pf);
-        em.add(u);
-        em.add(c);
-        em.add(p);
-        nf.addActionListener(this);
-        nf.setActionCommand("new");
-        sf.addActionListener(this);
-        sf.setActionCommand("save");
-        pf.addActionListener(this);
-        pf.setActionCommand("print");
-        c.addActionListener(this);
-
-        c.setActionCommand("copy");
-        p.addActionListener(this);
-        p.setActionCommand("paste");
-        u.addActionListener(this);
-        u.setActionCommand("undo");
-        mb.add(fm);
-        mb.add(em);
+        filesMenu.add(MenuItemUtility.GetMenuItem("Open", new OpenFileAction(textPane, recentFilesManager)));
+        filesMenu.add(MenuItemUtility.GetMenuItem("New File", new NewFileAction(textPane)));
+        filesMenu.add(MenuItemUtility.GetMenuItem("Save File", new SaveFileAction(textPane, recentFilesManager)));
+        filesMenu.add(MenuItemUtility.GetMenuItem("Print File", new PrintFileAction(textPane)));
+        filesMenu.add(recentFilesMenu);
+        editMenu.add(MenuItemUtility.GetMenuItem("Replace", new ReplaceAction(textPane)));
+        editMenu.add(MenuItemUtility.GetMenuItem("Copy", new CopyAction(textPane)));
+        editMenu.add(MenuItemUtility.GetMenuItem("Paste", new PasteAction(textPane)));
+        mb.add(filesMenu);
+        mb.add(editMenu);
         setJMenuBar(mb);
-        add(new JScrollPane(d));
+        add(new JScrollPane(textPane));
         setPreferredSize(new Dimension(600,600));
         setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
         setVisible(true);
         pack();
     }
+
     public static void main(String[] args) {
         SimpleNotePad app = new SimpleNotePad();
     }
-    @Override
```

Comment:
This was definitely the most ugly part of the original code. Removal was a good idea.